includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -155,6 +156,11 @@ public function __construct() {
 
 		foreach ( self::UPE_AVAILABLE_METHODS as $payment_method_class ) {
 
+			// Show ACH only if feature is enabled.
+			if ( WC_Stripe_UPE_Payment_Method_ACH::class === $payment_method_class && ! WC_Stripe_Feature_Flags::is_ach_lpm_enabled() ) {
```

Comment:
I saw that Bacs above uses a different approach to checking if the method should be enabled / added to the list. I'd like if we could keep the approach standardized (as much as possible). I'd prefer the approach in this PR just because it matches what's been used for Sofort and Giropay, but I'd be curious which is the preferred approach between @asumaran and @rafaelzaleski. Depending on which is preferred, this PR may not need changes and instead a change for Bacs would be required.

Reply:
Additionally, I was wondering if it's possible / preferrable to add an "enabled" property to the payment method class that would check against the feature flag? This could eliminate the need to check which class we're working on in the loop and abstract things a bit. I don't think it's necessary for this PR, but wondering if this would be desirable as a future enhancement?

Reply:
I did the same approach https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3775/files#diff-9356f87eb3d1b3fbc63a5201e8ccc38668d392af16d66e94656a3ee8e101820e

Reply:
Sorry I just discussed with @asumaran and the first comment is already addressed [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3727#issuecomment-2619531736). We can disregard the first comment then.

Reply:
> Additionally, I was wondering if it's possible / preferrable to add an "enabled" property to the payment method class that would check against the feature flag?

We could override the method `is_enabled_at_checkout` ([link](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/39c7201b55fd2731375fd89402a8ef838257632a/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L1662)) and add a check for the feature flag. However, I’d prefer not to load the payment method class at all while it's behind feature flags.